osu.Game/Rulesets/UI/ModIcon.cs

```diff
@@ -207,6 +235,11 @@ private void updateMod(IMod value)
             backgroundColour = colours.ForModType(value.Type);
             updateColour();
 
+            if (mod.HasNonDefaultSettings)
+                adjustmentMarker.Show();
+            else
+                adjustmentMarker.Hide();
```

Comment:
(This is the only real blocker) Case in previous point re: safety of reading this, this should be in the "update extended info" callback as it's hooked up to a change tracker:

https://github.com/ppy/osu/blob/3ee8eb2809b7570d315e823851214e47e9d64ba7/osu.Game/Rulesets/UI/ModIcon.cs#L189